lib/cli.nix

```
@@ -111,38 +114,219 @@ rec {
     :::
   */
   toGNUCommandLine =
-    {
-      mkOptionName ? k: if builtins.stringLength k == 1 then "-${k}" else "--${k}",
+    lib.warn
+      "lib.cli.toGNUCommandLine is deprecated, please use lib.cli.toCommandLine or lib.cli.toCommandLineShellGNU instead."
+      (
+        {
+          mkOptionName ? k: if builtins.stringLength k == 1 then "-${k}" else "--${k}",
+
+          mkBool ? k: v: lib.optional v (mkOptionName k),
+
+          mkList ? k: v: lib.concatMap (mkOption k) v,
+
+          mkOption ?
+            k: v:
+            if v == null then
+              [ ]
+            else if optionValueSeparator == null then
+              [
+                (mkOptionName k)
+                (lib.generators.mkValueStringDefault { } v)
+              ]
+            else
+              [ "${mkOptionName k}${optionValueSeparator}${lib.generators.mkValueStringDefault { } v}" ],
+
+          optionValueSeparator ? null,
+        }:
+        options:
+        let
+          render =
+            k: v:
+            if builtins.isBool v then
+              mkBool k v
+            else if builtins.isList v then
+              mkList k v
+            else
+              mkOption k v;
+
+        in
+        builtins.concatLists (lib.mapAttrsToList render options)
+      );
 
-      mkBool ? k: v: lib.optional v (mkOptionName k),
+  /**
+    Similar to `toCommandLineGNU`, with the exception that this returns a string of shell-escaped arguments instead of a list.
+  */
+  toCommandLineShellGNU =
+    options: attrs: lib.escapeShellArgs (lib.cli.toCommandLineGNU options attrs);
 
-      mkList ? k: v: lib.concatMap (mkOption k) v,
+  /**
+    Convert an attribute set into a list of GNU-style command line options.
```

```suggestion
    Converts an attribute set into a list of GNU-style command line options.
```
Please use present-tense :) https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#writing-function-documentation